src/Sylius/Component/Promotion/Generator/CouponGenerator.php

```diff
@@ -68,13 +72,19 @@ public function generate(PromotionInterface $promotion, Instruction $instruction
     /**
      * {@inheritdoc}
      */
-    public function generateUniqueCode()
+    public function generateUniqueCode($codeLength)
     {
         $code = null;
 
+        if (1 > $codeLength || 40 < $codeLength) {
```

Comment:
This must numeric or integer, so in case somone don't use form you should validated this. Also IMO this should land in model not here.